test/sampling_tests.cpp

```diff
@@ -116,12 +116,16 @@ void CreateRandomLogits(float* logits, int num_large, int vocab_size, int batch_
       logits[v + b * vocab_size] = dist(engine);
     }
   }
+
   // Randomly set num_large elements to be large
   std::uniform_int_distribution<> dist_large(0, vocab_size - 1);
   for (int b = 0; b < batch_size; b++) {
     for (int i = 0; i < num_large; i++) {
-      int index = dist_large(engine);
-      logits[index + b * vocab_size] = 25.0f;
+      float& value = logits[dist_large(engine) + b * vocab_size];
+      if (value == 25.0f)
+        i--;  // We hit the same number twice, so do it again to ensure num_large values are set to 25.0f
```

Comment:
Can we have a sanity check to make sure that `num_large` < logits.size(). Else, we will run into an infinite loop.

Reply:
could also increment index until reaching an unused one